### PR TITLE
execute CoreBuild to generate xaml files

### LINKED ISSUE #2779: MSBuildWorkspace and WPF Projects
I'm trying to compile a WPF project using `MSBuildWorkspace`. Should this work out of the box, or do I need to do anything special to set it up?

When I open a .sln containing a WPF project and try to compile I receive two errors:

```
C:\Users\Josh\Documents\GitHub\Orange\WpfApplication1\WpfApplication1\MainWindow.xaml.cs(25,13): error CS0103: The name 'InitializeComponent' does not exist in the current context
```
and

```
error CS5001: Program does not contain a static 'Main' method suitable for an entry point
```

To reproduce, simply create a new WPF project. (Mine was WPFApplication1.sln). Then just load into a workspace and try to compile the project.

```CSharp
var msws = MSBuildWorkspace.Create(new Dictionary<string, string> { { "CheckForSystemRuntimeDependency", "true" } });
var soln = msws.OpenSolutionAsync(@"C:\Users\Josh\Documents\GitHub\Orange\WpfApplication1\WpfApplication1.sln").Result;
var comp = soln.Projects.First().GetCompilationAsync().Result;
var errs = comp.GetDiagnostics().Where(n => n.Severity == DiagnosticSeverity.Error).ToList();
//Two errors :(
```

### PR DESCRIPTION
**Pull request by mattwar**
Fixes #2779 

This enables the MSBuild workspace to run the xaml tasks that generate files, so those files get added as compiler source inputs correctly. 

The solution is to switch to using the "CoreBuild" msbuild target instead of "Compile" (which is included in CoreBuild.)

Note: there has been lots of conversation on how/why the VS project system is able to use only the Compile target and still build/produce the correct compiler inputs, and the result is no one either knows are assumes there is project-system flavoring extensions augmenting the project system.  What is known, however, is that under normal msbuild-only conditions, the extra tasks needed to process xaml only occur under CoreBuild.  

So I'm switching msbuild workspace to use CodeBuild. The down side is that some extra/unnecessary work may happen when opening some projects.

@Pilchie @jasonmalinowski please review

*Also, this solution doesn't solve the versioning problem that requires apps using msbuild workspace to use .net config file redirects for msbuild API dlls, the same way the msbuild.exe and devenv.exe do.

### CODE REVIEW

File: src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/BuildTargets.cs [hunk 1]
```diff
@@ -32,6 +32,54 @@ public string[] Targets
             get { return _buildTargets.ToArray(); }
         }
 
+#if DEBUG
```
Comment (Pilchie) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/BuildTargets.cs]: What is the purpose of this code? It doesn't seem to be called.

File: src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs [hunk 2]
```diff
@@ -54,8 +54,18 @@ public string GetPropertyValue(string name)
...
+            var executedProject = _loadedProject.CreateProjectInstance();
+
+            // if the project doesn't have a Compile target, then its not really a vb or c# project.
+            if (!executedProject.Targets.ContainsKey("Compile"))
+            {
+                return executedProject;
+            }
+
+            // prepare for building  -- use CoreBuild to get xaml generated files
+            var buildTargets = new BuildTargets(_loadedProject, "CoreBuild");
```
Comment (Pilchie) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs]: Is there a way to write tests for this?

File: src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/BuildTargets.cs (hunk 1 above)
Comment (mattwar) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/BuildTargets.cs]: It's for debugging the tasks that MSBuild will process.

File: src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs (hunk 2 above)
Comment (mattwar) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs]: Yes, I can add a MSBuild workspace test that opens a wpf/xaml app.

Comment (mattwar) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs]: There is already a test, but it is skipped due to redirect problem with MSBuild.  We cannot supply the correct redirects to xunit.

Comment (Pilchie) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs]: @KevinH-MS mentioned that you can add a `TestAssemblyName.dll.config` file, and xunit will apply it's contents when running tests for that assembly.  Does that work here?

**Comment by mattwar** [comment]
Looks like the bug causing the properties not to get set properly was what was causing this to be different than VS.  This was fixed in #2765, so no need to change to CoreBuild.

**Comment by Pilchie** [comment]
In other words, you should be able to work around this in RC by adding `DesignTimeBuild` and `BuildingInsideVisualStudio` properties into your `MSBuildWorkspace`.

**Pull request closed without merging**